src/safeposix/net.rs

```diff
@@ -43,11 +41,24 @@ pub const TCPPORT: bool = true;
 pub const UDPPORT: bool = false;
 
 impl NetMetadata {
+    fn port_in_use(&self, tup: &(interface::GenIpaddr, u16, PortType)) -> bool {
```

Comment:
Would this function be helpful to add to the creation functions or sockets? It could clean up some code -- not sure though.

Reply:
This code is specifically for the _get_available_x port because instead of just needing to check the used port set for that device, we also need to check the used_port_set for the catch-all device ::0 or 0.0.0.0. For other code, checking the used_port_set suffices and is reasonably clean.